pkg/gitparse/gitparse.go

```diff
@@ -134,7 +135,24 @@ func (c *Parser) RepoPath(ctx context.Context, source string, head string, abbre
 	cmd := exec.Command("git", args...)
 	absPath, err := filepath.Abs(source)
 	if err == nil {
-		cmd.Env = append(cmd.Env, fmt.Sprintf("GIT_DIR=%s", filepath.Join(absPath, ".git")))
+		if !isBare {
+			cmd.Env = append(cmd.Env, fmt.Sprintf("GIT_DIR=%s", filepath.Join(absPath, ".git")))
+		} else {
+			cmd.Env = append(cmd.Env, fmt.Sprintf("GIT_DIR=%s", absPath))
+
+			// We need those variables to handle incoming commits
+			// while using trufflehog in pre-receive hooks
```

Comment:
@L11R **suggestion**: Rather than iterating over all the environment variables we should attempt to add them to `cmd.Env` with `os.Getenv`

Reply:
@zricethezav should be better now! Also replaced fmt.Sprintf with simple concatenation, it should be more performant.